src/utils/buffer.ts

```
@@ -12,3 +14,134 @@ export const hexToBuffer = (value: string): Buffer => {
   }
   return Buffer.from(value, 'hex');
 };
+
+/**
+ * Unpacks a buffer size
+ *
+ * @param {number} n The size of the buffer to unpack
+ * @param {Buffer} buff The buffer to unpack
+ *
+ * @return {[Buffer, Buffer]} The unpacked buffer followed by the rest of the buffer
+ */
+export const unpackLen = (n: number, buff: Buffer): [Buffer, Buffer] => {
+  return [
+    buff.slice(0, n),
+    buff.slice(n)
+  ];
+};
+
+/**
+ * Unpacks an integer from a buffer
+ *
+ * @param {number} n The size of the number in bytes
+ * @param {boolean} signed If the number is signed
+ * @param {Buffer} buff The buffer to unpack
+ *
+ * @return {[number, Buffer]} The unpacked number followed by the rest of the buffer
+ */
+export const unpackToInt = (n: number, signed: boolean, buff: Buffer): [number, Buffer] => {
+  let retInt;
+  const slicedBuff = buff.slice(0, n);
```

Maybe throw an exception if `n > len(buff)`?

---

Done.